docker/rucio_client/scripts/updateDDMQuota

```diff
@@ -90,52 +73,93 @@ def calculate_ddm_quotas():
-        if "override_ddm_quota" in rse_attributes:
-            overridden_ddm_quotas[rse] = rse_attributes["override_ddm_quota"]
-            continue
 
-        ddm_quotas[rse] = ddm_quota
-
-
-def normalize_ddm_quotas():
-    weights = [value for value in ddm_quotas.values()]
-    for rse, weight in ddm_quotas.items():
-        ddm_quotas[rse] = int(((weight - min(weights)) / (max(weights) - min(weights))) * MAX_DDM_QUOTA)
+        free = static - rucio
+        dm_weight = static_weight * (static / total_static) + free_weight * \
+            (free / static) + expired_weight * (expired / static)
+        dm_weight_coefficient = 1
+        if "dm_weight_coefficient" in rse_attributes:
+            dm_weight_coefficient = float(rse_attributes["dm_weight_coefficient"])
+        dm_weight *= dm_weight_coefficient
+        if make_quadratic:
+            dm_weight = dm_weight ** 2
+
+        stats[rse] = {
+            "pledge": static,
+            "free": free,
+            "dm_weight_coefficient" : dm_weight_coefficient,
+        }
+
+        dm_weights[rse] = dm_weight
+    return dm_weights
+
+
+def normalize_dm_weights(dm_weights):
+    # TODO: Don't take 0 values into normalization
+    weights = [value for value in dm_weights.values()]
+    min_positive_weight = min([n for n in weights if n > 0])
+    for rse, weight in dm_weights.items():
+
+        # Don't take 0 weights into normalization, since they are manually overriden
+        if weight == 0:
+            dm_weight = weight
+        else:
+            # This normalization sets the lowest dm_weight to 0
+            dm_weight = int(((weight - min_positive_weight) / (max(weights) - min_positive_weight)) * MAX_DM_WEIGHT)
+            #dm_weight = int((weight/sum(weights)) * MAX_DM_WEIGHT)
+        dm_weights[rse] = dm_weight
+        stats[rse]["dm_weight"] = dm_weight
 
+    return dm_weights
 
-def set_ddm_quotas():
-    # Set automatically calculated ddm quotas
-    for rse, ddm_quota in ddm_quotas.items():
-        if DRY_RUN:
-            print("DRY-RUN: Set ddm_quota for {} to {}".format(rse, ddm_quota))
-        else:
-            client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
-            print("Set ddm_quota for {} to {}".format(rse, ddm_quota))
 
-    # Set overriden ddm_quotas
-    for rse, ddm_quota in overridden_ddm_quotas.items():
+def set_dm_weights(dm_weights):
+    # Set automatically calculated dm weights
+    for rse, dm_weight in dm_weights.items():
         if DRY_RUN:
-            print("DRY-RUN: Override ddm_quota for {} to {}".format(rse, ddm_quota))
+            print("DRY-RUN: Set dm_weight for {} to {}".format(rse, dm_weight))
         else:
-            client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
-            print("Override ddm_quota for {} to {}".format(rse, ddm_quota))
-
+            client.add_rse_attribute(rse, "dm_weight", dm_weight)
+            # Keep setting ddm_quota for disk rses until we deprecate it
```

Comment:
Please remember to inform @belforte about it. 
In fact, I believe we should draft a mail now itself.

Reply:
Yes, it doesn't harm to inform early. Last time we changed anything with `ddm_quota`; it was a bit late. 
CRAB would need to make changes to their code as well.

Reply:
Sounds good, let's do it in the context of this issue: https://github.com/dmwm/CMSRucio/issues/772